### DISCUSSION TITLE
nullable DB fields

### DISCUSSION THREAD

**Discussion by j0sh** [pocketbase/pocketbase#4002] [Q&A] [answered]
Hello, related to https://github.com/pocketbase/pocketbase/issues/122 - it would be nice to have nullable fields in a collection. Right now, it is not possible to set a UNIQUE constraint on a new column in an existing table, due to the [`TEXT DEFAULT '' NOT NULL`](https://github.com/pocketbase/pocketbase/blob/1208edec92905bb661d9d57f3c9407b69a4559ba/models/schema/schema_field.go#L160C11-L160C35) definition for text fields, because multiple rows with a (default) empty string would violate the constraint. In SQLite, it is allowed to have NULL / unset rows while maintaining uniqueness among those that are set ([link](https://sqlite.org/faq.html#q26)).

There are various ways to work around this with pocketbase migrations, but nullable fields would make this a lot easier to deal with.

Maintaining the zero-value representation within golang and the API would probably still be OK in my case; it is just the underlying columns that should be null if unset (or if the golang / API are zero-valued).

**Comment by ganigeorgiev** [comment, marked as answer]
>  Right now, it is not possible to set a UNIQUE constraint on a new column in an existing table

You can create a [partial index](https://www.sqlite.org/partialindex.html), aka. specifying a `WHERE` clause. Something like:

```sql
CREATE UNIQUE INDEX idx_something ON example (title) WHERE title != ""
```

**Reply by j0sh** [reply]
That did the trick - thank you!
